server/src/main/java/org/spine3/server/storage/memory/InMemoryRecordStorage.java

```diff
@@ -64,24 +64,31 @@ protected InMemoryRecordStorage(boolean multitenant) {
     @SuppressWarnings("MethodWithMultipleLoops")    /* It's OK for in-memory implementation
                                                      * as it is used primarily in tests. */
     @Override
-    protected Iterable<EntityStorageRecord> readBulkInternal(final Iterable<I> givenIds, @Nullable FieldMask fieldMask) {
+    protected Iterable<EntityStorageRecord> readBulkInternal(final Iterable<I> givenIds, FieldMask fieldMask) {
         final Map<I, EntityStorageRecord> storage = getStorage();
 
         // It is not possible to return an immutable collection, since {@code null} may be present in it.
         final Collection<EntityStorageRecord> result = new LinkedList<>();
 
+        TypeUrl typeUrl = null;
+
         for (I recordId : storage.keySet()) {
             for (I givenId : givenIds) {
                 if (recordId.equals(givenId)) {
                     final EntityStorageRecord.Builder matchingRecord = storage.get(recordId).toBuilder();
                     final Any state = matchingRecord.getState();
 
-                    matchingRecord.setState(
-                            AnyPacker.pack(
-                                    FieldMasks.applyIfValid(
-                                            fieldMask,
-                                            AnyPacker.unpack(state),
-                                            TypeUrl.of(state.getTypeUrl()))));
+                    if (typeUrl == null) {
+                        typeUrl = TypeUrl.of(state.getTypeUrl());
+                    }
+
+                    final Any processed = AnyPacker.pack(
+                            FieldMasks.applyIfValid(
```

Comment:
Please split into several expressions.
The criterion is to have an exact line in case of unexpected `Throwable`. Right now it is impossible to trace which expression caused the failure.